### DISCUSSION TITLE
disambiguating-types-with-the-same-name-with-extern-alias/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#429] [Announcements]
# disambiguating-types-with-the-same-name-with-extern-alias/

In this post I describe how to solve Error CS0433, where you have two types with the exact same name and namespace coming from two different packages

https://andrewlock.net/disambiguating-types-with-the-same-name-with-extern-alias/

**Comment by GitClickOk** [comment]
Nice article. I have been working with that, and I want to leave a tip. What if the conflicting types are in transient libraries, ie, you are not referencing directly via PackageReference or ProjectReference? The answer is, discover the referenced transient package name, and assign the alias like this:
```xml
<ItemGroup>
	<Reference Include="SomeTransientClassLibrary">
		<Aliases>MyTransientAlias<Aliases>
	</Reference>
</ItemGroup>
```
I got a hard time to find this info, so I think important to mention this here.

Best Regards!

**Comment by olivermue** [comment]
Had the same problem in the past, when the .Net Microsoft Graph SDK and Beta SDK shared the same types in the same namespaces (but thankfully this has been changed). The next problem comes into play, when you try to create a test project for your main project. The aliases projects have to be referenced again in the test project and defining the aliases on themselves. Otherwise you run in the same issue again. This can get really cumbersome if you have big solutions with deep hierarchies with 5 or more levels, because you have to add the reference with the alias on each level if any type is used within the project.
